packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt

```diff
@@ -110,6 +174,188 @@ class FileDownloader(
     }
   }
 
+  private fun isPatchResponse(responseBody: ResponseBody): Boolean {
+    val mediaType: MediaType? = responseBody.contentType()
+    return mediaType?.let { it.type == "application" && it.subtype == "vnd.bsdiff" } ?: false
+  }
+
+  @Throws(IOException::class)
+  private fun applyPatchAndVerify(
+    asset: AssetEntity,
+    request: Request,
+    responseBody: ResponseBody,
+    destination: File,
+    updatesDirectory: File,
+    expectedBase64URLEncodedSHA256Hash: String?
+  ): ByteArray {
+    val launchAssetContext = prepareAssetForDiff(
+      asset,
+      request,
+      responseBody,
+      updatesDirectory
+    )
+
+    val requestedUpdateId = request.header(EXPO_REQUESTED_UPDATE_ID_HEADER)
+
+    return responseBody.use { body ->
+      applyHermesDiff(
+        baseFile = launchAssetContext.baseFile,
+        diffBody = body,
+        destination = destination,
+        expectedBase64URLEncodedSHA256Hash = expectedBase64URLEncodedSHA256Hash,
+        asset = asset,
+        requestedUpdateId = requestedUpdateId
+      )
+    }
+  }
+
+  private data class LaunchAssetContext(val baseFile: File)
+
+  private fun prepareAssetForDiff(
+    asset: AssetEntity,
+    request: Request,
+    responseBody: ResponseBody,
+    updatesDirectory: File
+  ): LaunchAssetContext {
+    if (!asset.isLaunchAsset) {
+      failWithClosedBody(responseBody, "Received Hermes diff for non-launch asset ${asset.key}")
```

Comment:
```suggestion
      failWithClosedBody(responseBody, "Received patch for non-launch asset ${asset.key}")
```

(or "diff", not sure which is more correct)